crates/lowering/src/ast_utils.rs

```diff
@@ -0,0 +1,969 @@
+                    attr,
+                },
+                Expr::BinOperation { left, op, right } => Expr::BinOperation {
+                    left: Box::new(map_ast_node((*left).into(), map_fn).as_expr()),
+                    right: Box::new(map_ast_node((*right).into(), map_fn).as_expr()),
+                    op,
+                },
+                Expr::BoolOperation { left, op, right } => Expr::BoolOperation {
+                    left: Box::new(map_ast_node((*left).into(), map_fn).as_expr()),
+                    right: Box::new(map_ast_node((*right).into(), map_fn).as_expr()),
+                    op,
+                },
+                Expr::Call {
+                    args,
+                    func,
+                    generic_args,
+                } => Expr::Call {
+                    args: map_call_args(args, map_fn),
+                    func: Box::new(map_ast_node((*func).into(), map_fn).as_expr()),
+                    generic_args,
+                },
+                Expr::CompOperation { left, op, right } => Expr::CompOperation {
+                    left: Box::new(map_ast_node((*left).into(), map_fn).as_expr()),
+                    right: Box::new(map_ast_node((*right).into(), map_fn).as_expr()),
+                    op,
+                },
+                Expr::List { elts } => Expr::List {
+                    elts: elts
+                        .into_iter()
+                        .map(|val| map_ast_node(val.into(), map_fn).as_expr())
+                        .collect(),
+                },
+                Expr::Subscript { value, index } => Expr::Subscript {
+                    value: Box::new(map_ast_node((*value).into(), map_fn).as_expr()),
+                    index: Box::new(map_ast_node((*index).into(), map_fn).as_expr()),
+                },
+                Expr::Ternary {
+                    if_expr,
+                    test,
+                    else_expr,
+                } => Expr::Ternary {
+                    if_expr: Box::new(map_ast_node((*if_expr).into(), map_fn).as_expr()),
+                    test: Box::new(map_ast_node((*test).into(), map_fn).as_expr()),
+                    else_expr: Box::new(map_ast_node((*else_expr).into(), map_fn).as_expr()),
+                },
+                Expr::Tuple { elts } => Expr::Tuple {
+                    elts: elts
+                        .into_iter()
+                        .map(|val| map_ast_node(val.into(), map_fn).as_expr())
+                        .collect(),
+                },
+                Expr::UnaryOperation { op, operand } => Expr::UnaryOperation {
+                    op,
+                    operand: Box::new(map_ast_node((*operand).into(), map_fn).as_expr()),
+                },
+                // The following *could* be covered via catch all. However, that would turn into a footgun if we add
+                // more expressions in the future that need to be walked. It's better to not use a catch all here.
```

Comment:
agreed. we can shorten this up with `|` separates, tho. still breaks the match statement when the enum is updated and saves us from having to repeat `expr.kind`

Reply:
Ah, of course!